### PR TITLE
FIX authorize catchall pricelist item

### PR DESCRIPTION
**Pull request by bealdav**
include in the commit 

catchall management and fix for new pylint tests

### CODE REVIEW

File: pricelist_per_product/pricelist.py [hunk 1]
```diff
@@ -126,8 +128,39 @@ def create(self, vals):
...
-            if 'product_id' in vals:
+            if not self._is_catch_all_item(vals) and 'product_id' in vals:
                 vals['related_sequence'] = item._get_sequence_price_grid(
                     vals)
             super(ProductPricelistItem, item).write(vals)
         return True
+
+    @api.model
+    def _is_catch_all_item(self, vals):
+        """ Sale Pricelists require a pricelist item in order to works fine.
```
Comment (hparfr) [inline pricelist_per_product/pricelist.py]: work

File: pricelist_per_product/pricelist.py [hunk 2]
```diff
@@ -126,8 +128,39 @@ def create(self, vals):
...
+            This item must be based on public price without discount/surcharge
+            If not, public price from product is not propagated in sale line
+            and used price is 0.0
```
Comment (hparfr) [inline pricelist_per_product/pricelist.py]: rm 'used'

**Review by hparfr — COMMENTED** [review]
minor remarks in comments

**Review by sebastienbeau — APPROVED** [review]
Indeed this method is really need, code review no test

**Pull request merged**